### DISCUSSION TITLE
How to limit the current through the client_id (custom) of the request header

### DISCUSSION THREAD

**Discussion by ouyangli** [apache/apisix#6918] [Q&A] [answered]
In my business, I need to limit the flow through the client-id in the request header, which stipulates that each client-id can only send 2 requests within 15 seconds

The configuration is as follows：
![image](https://user-images.githubusercontent.com/8622409/164971105-81eec21d-4722-44d9-a2c6-08e4277f8930.png)

called by postman:
![image](https://user-images.githubusercontent.com/8622409/164971053-26943e51-5186-4e72-931d-6fdbeb6999da.png)

Expected effect: Each client-id can only send 2 requests per 15 seconds

Actual effect: all client-id can only send 2 requests every 15 seconds

**Comment by tzssangglass** [comment, marked as answer]
see: https://github.com/apache/apisix/issues/6656#issuecomment-1073938173

**Reply by ouyangli** [reply]
No custom variables are required, refer to the following information:

reference： https://github.com/apache/apisix/blob/master/docs/en/latest/admin-api.md#upstream

Related Instructions：

The following should be considered when setting the hash_on value:

When set to vars, a key is required. The value of the key can be any of the [Nginx variables](http://nginx.org/en/docs/varindex.html) without the $ prefix.
When set to header, a key is required. This is equal to "http_key".
When set to cookie, a key is required. This key is equal to "cookie_key". The cookie name is case-sensitive.
When set to consumer, the key is optional and the key is set to the consumer_name captured from the authentication Plugin.
When set to vars_combinations, the key is required. The value of the key can be a combination of any of the [Nginx variables](http://nginx.org/en/docs/varindex.html) like $request_uri$remote_addr.
When no value is set for either hash_on or key, the key defaults to remote_add



![image](https://user-images.githubusercontent.com/8622409/165044681-c7858b52-9fb3-487f-ac58-2b92dc55b036.png)
